io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -635,17 +634,16 @@ private void writePropertiesInstanceData(
 					&& prop.getName().getLocalPart() != null) {
 				Object value = new InstanceAccessor(instance)
 						.findChildren(prop.getName().getLocalPart()).value();
-				List<GeometryProperty<?>> geoms = traverseInstanceForGeometries(instance);
-				// add value by traversing geometryType from instance
-				for (GeometryProperty<?> geoProp : geoms) {
-					if (geoProp.getGeometry() != null) {
-						String geometryType = geoProp.getGeometry().getGeometryType();
-						if (schemaFbMap.get(localPart) != null
-								&& schemaFbMap.get(localPart).get(geometryType) != null) {
-							schemaFbMap.get(localPart).get(geometryType).add(value);
+				if (schemaFbMap.get(localPart) != null && value != null) {
```

Comment:
@emanuelaepure10 as discussed I think a null check for `value` here might have impact on attributes not present in the export. Also, this might not prevent the IndexOutOfBounds exception if all the additional values are not null. I believe we should analyse the data and try to pass in all the attributes during schema creation in step 2.